tests/test_conditional_file_name.py

```diff
@@ -55,3 +57,34 @@ def test_dont_render_conditional_subdir(tmp_path_factory: TempPathFactory) -> No
     )
     copier.run_auto(str(src), dst)
     assert not (dst / "subdir" / "file.txt").exists()
+
+
+def test_answer_changes(tmp_path_factory: TempPathFactory) -> None:
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+
+    with local.cwd(src):
+        build_file_tree(
+            {
+                "{{ _copier_conf.answers_file }}.jinja": "{{ _copier_answers|to_nice_yaml }}",
+                "copier.yml": """
+                    condition:
+                        type: bool
+                """,
+                "{% if condition %}file.txt{% endif %}.jinja": "",
+            }
+        )
+        git("init")
+        git("add", ".")
+        git("commit", "-mv1")
+        git("tag", "v1")
+
+    copier.copy(str(src), dst, data={"condition": True})
+    assert (dst / "file.txt").exists()
+
+    with local.cwd(dst):
+        git("init")
+        git("add", ".")
+        git("commit", "-mv1")
+
+    copier.copy(dst_path=dst, data={"condition": False})
+    assert not (dst / "file.txt").exists()
```

Comment:
I have tested this without the fix and it passes ✔️. Maybe some of the recent fixes fixed this already as a side effect?

If so, we could merge the test only and it should be enough. Or maybe the test isn't correct?

Reply:
I think you're right that the recent fixes fixed also this problem as a side effect. I'll comment more below.